docs/chain-abstraction/omnibridge/implementation.md

```diff
@@ -0,0 +1,257 @@
+    Locked --> Signed: MPC signature generated
+    Signed --> Completed: Tokens released on Ethereum
+    Completed --> [*]
+```
+
+---
+
+## Message Signing and Verification
+
+For most chains, the bridge uses a payload-based message signing system (with Bitcoin being a notable exception requiring full transaction signing).
+
+### Message Types
+
+The bridge supports several types of signed messages:
+
+* **Transfer Messages**
+  * Initiation messages
+  * Finalization messages
+* **Token Messages**
+  * Deployment messages
+  * Metadata update messages
+
+### Payload Structure
+
+Messages are encoded using Borsh serialization and contain:
+
+| Component | Description |
+|-----------|-------------|
+| Message Type | Identifier for the message category |
+| Chain Info | Chain IDs and relevant addresses |
+| Operation Data | Amounts, recipients, fees, etc. |
+
+### Signature Process
+
+1. NEAR contract creates and stores the message payload
+2. MPC network observers detect valid payloads
+3. Nodes jointly sign the payload
+4. Signature is verified on destination chains
+
+:::tip Key Benefits
+* Clearer message intent through structured payloads
+* More efficient signature verification on destination chains
+* Standardized message format across chains
+:::
+
+## Transaction Flow: NEAR to Other Chains
+
+Here's an overview of how transfers are processed from NEAR to different destination chains:
+
+```mermaid
+flowchart TD
+    Start[User Initiates Transfer] --> TokenCheck{Token Type?}
+    
+    TokenCheck -->|NEAR Native| Lock[Lock in Bridge Contract]
+    TokenCheck -->|Bridged Token| Burn[Burn Token]
+    
+    Lock --> MPCSign[Request MPC Signature]
+    Burn --> MPCSign
+    
+    MPCSign --> Chain{Destination Chain}
+    
+    Chain -->|Ethereum| EVMBridge[EVM Bridge Contract]
+    Chain -->|Bitcoin| BTCBridge[Bitcoin Script]
+    Chain -->|Other| WormBridge[Wormhole Bridge]
+    
+    EVMBridge --> Mint[Mint/Release Tokens]
+    BTCBridge --> Mint
+    WormBridge --> Mint
+    
+    Mint --> End[Transfer Complete]
+```
+
+### Transfer Process
+
+Let's follow what happens when a user wants to transfer tokens from NEAR to another chain:
+
+#### 1. Initiation
+
+The user starts by calling the token contract with:
+
+* Amount to transfer
+* Destination chain and address
+* Fee preferences (whether to pay fees in the token being transferred or in NEAR)
+
+#### 2. Token Lock
+
+The token contract transfers tokens to the locker contract, which:
+
+* Validates the transfer message
+* Assigns a unique nonce
+* Records the pending transfer
+* Emits a transfer event
+
+#### 3. MPC Signing
+
+The bridge contract:
+
+* Requests signature generation
+* MPC nodes jointly generate and aggregate signature
+* Maintains threshold security throughout process
+
+#### 4. Destination Chain
+
+The Bridge Token Factory on the destination chain:
+
+* Verifies the MPC signature
+* Mints equivalent tokens
+* Fees are minted on NEAR side for relayers
```

Comment:
I feel like this does not belong under #4 / Destination Chain, but somewhere earlier, either #1 or #2.